presto-parquet/src/main/java/io/prestosql/parquet/predicate/PredicateUtils.java

```diff
@@ -161,20 +162,22 @@ private static boolean isColumnPredicate(ColumnDescriptor columnDescriptor, Tupl
 
     @VisibleForTesting
     @SuppressWarnings("deprecation")
-    public static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings)
+    static boolean isOnlyDictionaryEncodingPages(ColumnChunkMetaData columnMetaData)
     {
-        // TODO: update to use EncodingStats in ColumnChunkMetaData when available
+        EncodingStats stats = columnMetaData.getEncodingStats();
+        if (stats != null) {
```

Comment:
need to check stats.usesV2Pages()

Reply:
I think this check may work either way if `EncodingStats` is available. Let me try out using a version of parquet-mr that has this change available (I think 1.9.0+) but then force the V1 format and see if writes out encoding stats but with plain dictionary (seems like maybe encoding stats isn't technically a v2 only feature given it has that flag in regards to if v2 pages were used).

Reply:
I confirmed if a version of the Parquet writer is used that knows about page encoding stats then it will write these even if configuring the writer for the V1 format. So if encoding stats are available this should work whether the write was for the V2 or V1 format.